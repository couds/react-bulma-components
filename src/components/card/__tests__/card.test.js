import React from 'react';
import { render } from 'react-testing-library';
import { Card } from '..';

describe('Card component', () => {
  it.each([[Card], [Card.Content], [Card.Footer], [Card.Header]])('should render', Component => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render card-image', () => {
    const { asFragment } = render(<Card.Image size="4by3" src="http://bulma.io/images/placeholders/1280x960.png" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render card-header', () => {
    const { asFragment } = render(
      <Card>
        <Card.Header>
          <Card.Header.Title>Title</Card.Header.Title>
          <Card.Header.Icon>
            <i className="icon" />
          </Card.Header.Icon>
        </Card.Header>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render card-footer', () => {
    const { asFragment } = render(
      <Card>
        <Card.Footer>
          <Card.Footer.Item>Yes</Card.Footer.Item>
          <Card.Footer.Item>No</Card.Footer.Item>
        </Card.Footer>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
