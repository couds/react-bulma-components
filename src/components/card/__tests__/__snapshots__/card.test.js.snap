// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should have card classname 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    Card Content
  </div>
</DocumentFragment>
`;

exports[`Card component Should have card-content classname 1`] = `
<DocumentFragment>
  <div
    class="card-content"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Card component Should have card-footer's classname 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-footer"
    >
      <div
        class="card-footer-item"
      >
        Yes
      </div>
      <div
        class="card-footer-item"
      >
        No
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component Should have card-header's classname 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      <div
        class="card-header-title"
      >
        Title
      </div>
      <div
        class="card-header-icon"
      >
        <i
          class="icon"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component Should have card-image classname 1`] = `
<DocumentFragment>
  <div
    class="card-image"
  >
    <figure
      class="image is-4by3"
    >
      <img
        alt=""
        class=""
        src="http://bulma.io/images/placeholders/1280x960.png"
      />
    </figure>
  </div>
</DocumentFragment>
`;
